sinopia_indexer/index.js

```diff
@@ -0,0 +1,41 @@
+// Copyright 2019 Stanford University see Apache2.txt for license
+
+const elasticsearch = require('elasticsearch')
+const Stomp = require('stomp-client')
+const superagent = require('superagent')
+
+const searchHost = process.env.SEARCH_HOST || 'localhost'
+const brokerHost = process.env.BROKER_HOST || 'localhost'
+
+const searchClient = new elasticsearch.Client({
+  host: `${searchHost}:9200`,
+  log: 'trace'
+})
+const queueName = '/queue/trellis'
+
+function main() {
+  let stompClient = new Stomp(brokerHost, 61613)
+  console.log(`connecting to stomp at ${brokerHost}:61613`)
+  stompClient.connect((sessionId) => {
+    console.log(`subscribing to ${queueName}, waiting for messages`)
+    stompClient.subscribe(queueName, (body, headers) => {
+      console.log(`received message with body: ${body}`)
+      uri = JSON.parse(body).object.id
+      console.log(`uri needs indexing: ${uri}`)
+      superagent.get(uri)
+        .accept('application/ld+json')
+        .then(res => {
+          json = res.body
+          console.log(`indexing trellis resource: ${json}`)
+          let indexResult = searchClient.index({
```

Comment:
If the indexResult is unsuccessful; would we send it to console.error or somewhere else? (Do you know if AWS fargate containers would have their logs available?)

Reply:
@jermnelson I think `console.error` is the simplest choice here. I suspect the errors would be automagically directed to CloudWatch Logs, but we ought to verify this. (It was at least true when writing golang code in Lambdas that output from the built-in `log` package went to CW Logs.) Error handling is another thing that needs to be added to this branch. I would do that in the course of writing up the test suite.

Reply:
> I suspect the errors would be automagically directed to CloudWatch Logs, but we ought to verify this. (It was at least true when writing golang code in Lambdas that output from the built-in `log` package went to CW Logs.)

that would be cool if we got that for free.  i imagine that's an easier thing for amazon to wire up in a lambda, where they're more directly controlling the container that runs the code (which seems like it'd give tighter integration by default with the language and its packages).  but this is all new enough to me that i would be totally unsurprised if that sort of magic did exist.

Reply:
confirmed with @aaron-collier today that as long as the Fargate service/container is wired up to use AWS logging, STDOUT/STDERR from the container should flow to CW Logs.

Reply:
👍

Reply:
that's great!